### PR TITLE
PHPORM-232 Support whereLike and whereNotLike

### LINKED ISSUE #3080: [Feature Request]  Add whereLike clause to query builder
https://github.com/laravel/framework/pull/52147

### PR DESCRIPTION
**Pull request by GromNaN**
Fix PHPORM-232
Fix #3080

### Checklist

- [x] Add tests and ensure they pass
- [x] Add an entry to the CHANGELOG.md file
- [ ] Update documentation for new features

### CODE REVIEW

File: tests/Query/BuilderTest.php [hunk 1]
```diff
@@ -748,6 +748,36 @@ function (Builder $builder) {
...
+
+        yield 'whereNotLike case not sensitive' => [
+            ['find' => [['name' => ['$not' => new Regex('^1$', 'i')]], []]],
+            fn (Builder $builder) => $builder->whereNotLike('name', '1', false),
+        ];
+
+        yield 'whereNotLike case sensitive' => [
+            ['find' => [['name' => ['$not' => new Regex('^1$', '')]], []]],
+            fn (Builder $builder) => $builder->whereNotLike('name', '1', true),
+        ];
```
Comment (GromNaN) [inline tests/Query/BuilderTest.php]: Tests from the Laravel PR: https://github.com/laravel/framework/pull/52147/files#diff-ec9430099997706fe1610c5864d456b250709d17781ad3aaf9bd0efe40418a42R659

File: src/Query/Builder.php [hunk 2]
```diff
@@ -1324,6 +1325,13 @@ protected function compileWhereNotIn(array $where): array
         return [$where['column'] => ['$nin' => array_values($where['values'])]];
     }
 
+    protected function compileWhereLike(array $where): array
+    {
+        $where['operator'] = $where['not'] ? 'not like' : 'like';
```
Comment (GromNaN) [inline src/Query/Builder.php:1330]: From Laravel: https://github.com/laravel/framework/pull/52147/files#diff-7ea7f749a25a923af9ac94f5fabe101c9bbbc98e448f97eeed773da965042a8cR324

File: tests/Query/BuilderTest.php [hunk 3]
```diff
@@ -748,6 +753,48 @@ function (Builder $builder) {
             fn (Builder $builder) => $builder->where('name', 'like', '_ac__me_'),
         ];
 
+        yield 'whereLike' => [
+            ['find' => [['name' => new Regex('^1$', 'i')], []]],
+            fn
+        (Builder $builder) => $builder->whereLike('name', '1'),
```
Comment (alcaeus) [inline tests/Query/BuilderTest.php:759]: Nit: The indentation seems off in this data provider

File: tests/Query/BuilderTest.php [hunk 4]
```diff
@@ -1161,142 +1208,154 @@ function (Builder $elemMatchQuery): void {
...
-                [
-                    'find' => [
-                        [
-                            '$and' => [
-                                ['last_name' => ['$not' => new Regex('^.*Doe.*$', 'i')]],
-                                ['email' => ['$not' => new Regex('^.*Doe.*$', 'i')]],
-                            ],
+            ],
+            fn
+            (Builder $builder) => $builder->whereAll(['last_name', 'email'], 'Doe'),
```
Comment (alcaeus) [inline tests/Query/BuilderTest.php:1220]: Same indentation issue as above in this data provider

**Review by alcaeus — CHANGES_REQUESTED** [review]

File: tests/Query/BuilderTest.php (hunk 3 above)
Comment (GromNaN) [reply tests/Query/BuilderTest.php:759]: Phpcs does this 😥

Comment (alcaeus) [reply tests/Query/BuilderTest.php:759]: Oh wow. Didn't say a word then.

**Review by alcaeus — APPROVED** [review]

**Pull request merged**